### ISSUE TITLE
Sorting for alphabetization can lead to inconsistencies

### ISSUE THREAD

**Issue by Mekadrom** [DraqueT/PolyGlot#532]
Before correction, algorithm for sorting by custom order could lead to ambiguous situations (presumption of object preference for preposition if self pattern not found WITHOUT checking if comparison value matches a pattern).

USER FEEDBACK BELOW: I currently have 352 words in my lexicon. While trying to add a 353rd, the letters are typed backwards in the conlang word textbox. After attempting to make a 353rd word, I cannot save the language. Attempting to save causes a soft lock on everything other than the textboxes for the currently active word and the word list.

I have logographs defined, but they do not work. I don't know if this affects it. Running Windows 10, Polyglot version 2.2.

Here is what happens after adding the next word:

![image](https://user-images.githubusercontent.com/1545535/35774905-035acdbe-094b-11e8-9a7e-8f4a623d2c54.png)

The loading symbol appears as the mouse anywhere other than the lexicon list and the textboxes. I cannot change words. I typed 'backwards' but it ends up reversed. I assume no one else has this problem, so I am willing to answer any questions about my system/setup.

**Comment by DraqueT** [comment]
Would you mind sending me a copy of the language file you're working on? I
just tried to reproduce this with that number of words and am not seeing
the same bug. Looking at it directly would help track it down a lot!

Also, once I figure out what's going on there, I can send a beta version
with that particular bug fixed over to you. ^^

- Draque

**Comment by Mekadrom** [comment]
Would you need the logograph images? And how would you like the file hosted? I can't drop the file here.

**Comment by DraqueT** [comment]
If you let me know the steps to encounter the logograph problems you’re having, I won’t need those files. That part of the system underwent a bit of an overhaul recently, so I’m not surprised there are a few bugs. 

As for the language file, could you just email is to draquemail@gmail.com or share it on something like dropbox? Either would work pretty well for me.

Thanks for taking the time to let me know what issues you’re having! I can’t know what to fix without good user feedback. ^^

-Draque

**Comment by Mekadrom** [comment]
I'm going to go ahead and say it's probably my fault; I don't think I ever figured out how the logograph system worked. I assumed that I would add the images and their associations with characters or sounds, but it never seemed to work. I also figured out that if I added a word with random letters/descriptions, I could then add another word perfectly fine and then delete the broken one. I sent the email.
